adapter/src/main/java/com/twilio/kudu/sql/KuduRelNode.java

```diff
@@ -75,10 +79,12 @@ class Implementor {
     public long limit = -1;
     public long offset = -1;
     public boolean sorted = false;
+    // flag that determines if we are grouping and ordering by a prefix of the pk
+    // columns so that we can push down the sort
+    public boolean groupBySorted = false;
+    // flag that determines if we are grouping by a prefix of the pk columns so that
+    // we can push down the limit
     public boolean groupByLimited = false;
-    // if groupByLimited is true and sortPkPrefixColumns is empty
```

Comment:
`sortPkPrefixColumns` can be derived from `sortPkColumns` so removed it and just added a flag `groupByLimited`

Reply:
Just trying to educate myself and be sure - When both `groupBySorted` and `groupByLimited` is true, we are pushing down sort but sorting again after fetching rows ? 

Or is that both can never be true together ?

Reply:
They can't be true together. `groupBySorted` is true only if any of the subclasses of `KuduSortRule` are matched (meaning we can push down the sort and also do limit/offset if needed). 
`groupByLimited` is true only if `KuduAggregationLimitRule` is matched (eg doing an aggregation that sorts by date desc and some other non pk column). In this case we can't eliminate the sort but we can limit how many rows are read from kudu (and sort on the client).